### PR TITLE
Form validation

### PR DESCRIPTION
**Pull request by GeekyHacks**
In this pull request I did the following:
- Add validation attributes to form inputs
- Make a function for invalid inputs
- Restyle the form section
- Fix linter errors

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -148,13 +148,13 @@ <h3 class="skills-text">Skills</h3>
...
           <form method="post" autocomplete="off" action="https://formspree.io/f/xdovwjar" class="inputForm">
-            <input id="firstname-input" onchange="saveValue(this);" class="full_Name" type="text" placeholder="Full Name" maxlength="30" name="First-name" required />
-            <div class="fullName">
-              <input id="firstname-input" onchange="saveValue(this);" class="name" type="text" placeholder="First Name" maxlength="30" name="First-name" required />
-              <input id="secondname-input" onchange="saveValue(this);" class="name" type="text" placeholder="Second name" maxlength="30" name="Second-name" required />
-            </div>
+            <input id="fullName" onchange="saveValue(this);" class="full_Name" type="text" placeholder="Full Name" maxlength="30" name="First-name" required />
+            <!-- 
+            <input id="firstname-input" onchange="saveValue(this);" class="name" type="text" placeholder="First Name" maxlength="30" name="First-name" required />
+            <input id="secondname-input" onchange="saveValue(this);" class="name" type="text" placeholder="Second name" maxlength="30" name="Second-name" required /> -->
```
Comment (1GORDON) [inline index.html:154]: Great job team 🤝

- [ ] Kindly remove commented code from your project as this is regarded as a bad practice.

File: index.html [hunk 2]
```diff
@@ -148,13 +148,13 @@ <h3 class="skills-text">Skills</h3>
         <div class="contact-dev">
           <h2 class="contactmeHeader">I'm always interested in hearing about new projects, so if you'd like to chat please get in touch.</h2>
           <form method="post" autocomplete="off" action="https://formspree.io/f/xdovwjar" class="inputForm">
-            <input id="firstname-input" onchange="saveValue(this);" class="full_Name" type="text" placeholder="Full Name" maxlength="30" name="First-name" required />
-            <div class="fullName">
-              <input id="firstname-input" onchange="saveValue(this);" class="name" type="text" placeholder="First Name" maxlength="30" name="First-name" required />
-              <input id="secondname-input" onchange="saveValue(this);" class="name" type="text" placeholder="Second name" maxlength="30" name="Second-name" required />
-            </div>
+            <input id="fullName" onchange="saveValue(this);" class="full_Name" type="text" placeholder="Full Name" maxlength="30" name="First-name" required />
```
Comment (1GORDON) [inline index.html:151]: Great job team 🤝

- [ ] As stated in the project requirements, If a user submits the contact form with the email field containing uppercase characters an error message should show up. Kindly ensure an error message shows up when an email containing an uppercase letter is used.
- [ ] Also ensure the error message should show up close to the `submit` button or the `input fields`

File: JS/contactForm.js [hunk 3]
```diff
@@ -2,34 +2,35 @@
...
 // the input valdiation
 function InvalidMsg(input) {
   if (input.value === '') {
     input.setCustomValidity('Entering an email-id is necessary!');
   } else if (input.validity.patternMismatch) {
-    input.setCustomValidity('Please enter a lowercase email address!');
+    input.setCustomValidity('Please enter a correct lowercase email address!');
   } else {
     input.setCustomValidity('');
   }
```
Comment (1GORDON) [inline JS/contactForm.js:36]: Great job team 🤝

- [ ] Kindly check if the email is all lowercase by comparing if the email and its lowercase converted version are the same, ie: `email === email.toLowerCase()`.

File: JS/contactForm.js [hunk 4]
```diff
@@ -2,34 +2,35 @@
...
+//   return localStorage.getItem(v);
+// }
 
-// save input values to localstorage
-document.getElementById('firstname-input').value = getSavedValue('firstname-input');
-document.getElementById('secondname-input').value = getSavedValue('secondname-input'); // set the value to this input
-document.getElementById('email-input').value = getSavedValue('email-input');
-document.getElementById('textarea-input').value = getSavedValue('textarea-input');
+// // save input values to localstorage
+// document.getElementById('firstname-input').value = getSavedValue('firstname-input');
```
Comment (1GORDON) [inline JS/contactForm.js:22]: Great job team 🤝

- [ ] You may also remove this commented code from your project as it is regarded a bad practice.

**Review by 1GORDON — CHANGES_REQUESTED** [review]
##
## Hi @GeekyHacks, and @Bupilipili ✋

Good job so far!

## Highlights!
- Followed GitHub Flow ✔️
- Professional readme ✔️

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend considering them as they can improve your code._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@1GORDON` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: JS/contactForm.js (hunk 3 above)
Comment (GeekyHacks) [reply JS/contactForm.js:36]: I did it by checking a certain lowercase email pattern.

**Review by algerina — APPROVED** [review]
Hi @GeekyHacks  ,

Great job so far!  

May I highlight the following: 
- [x] Your  form is working correctly👏  ✔️ 
- [x] You implemented the requirements of the previous TSE✔️ 





## STATUS: APPROVED :muscle::clap:

Congratulations! :tada:

Your project is complete!  It's time to merge it :shipit:



 
 ## Optional suggestions

 - Please check the comment under the review.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._
Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread and tag me @algerina if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)
___
_If anything seems to be unclear to you, reach out and ask anything by commenting on this Pull Request and mention me as [Algerina](https://github.com/algerina), I will receive a notification and will respond._

[GitHub](https://github.com/algerina) | [LinkedIn](https://linkedin.com/in/amel-khiri) | [Twitter](https://twitter.com/AmelQahwadji)

**Pull request merged**